orchagent/fdborch.cpp

```diff
@@ -174,41 +173,160 @@ bool FdbOrch::storeFdbEntryState(const FdbUpdate& update)
     }
 }
 
+/*
+clears stateDb and decrements corresponding internal fdb counters
+*/
+void FdbOrch::clearFdbEntry(const MacAddress& mac,
+                   const sai_object_id_t& bv_id,
+                   const string& fdb_type,
+                   const string& port_alias)
+{
+    FdbUpdate update;
+    update.entry.mac = mac;
+    update.entry.bv_id = bv_id;
+    update.type = fdb_type;
+    update.add = false;
+
+    /* Fetch Vlan and decrement the counter */
+    Port temp_vlan;
+    if (m_portsOrch->getPort(bv_id, temp_vlan))
+    {
+        m_portsOrch->decrFdbCount(temp_vlan.m_alias, 1);
+    }
+
+    /* Decrement port fdb_counter */
+    m_portsOrch->decrFdbCount(port_alias, 1);
+
+    /* Remove the FdbEntry from the internal cache, update state DB and CRM counter */
+    storeFdbEntryState(update);
+    notify(SUBJECT_TYPE_FDB_CHANGE, &update);
+
+    SWSS_LOG_INFO("FdbEntry removed from internal cache, MAC: %s , port: %s, BVID: 0x%" PRIx64,
+                   mac.to_string().c_str(), port_alias.c_str(), bv_id);
+}
+
+/*
+Handles the SAI_FDB_EVENT_FLUSHED notification recieved from syncd
+Ref: https://github.com/opencomputeproject/SAI/blob/master/inc/saifdb.h#L223
+*/
+void FdbOrch::handleSyncdFlushNotif(const sai_object_id_t& bv_id,
+                                    const sai_object_id_t& bridge_port_id,
+                                    const string& fdb_type,
+                                    const MacAddress& mac)
+{
+    MacAddress flush_mac(CONSOLIDATED_FLUSH_MAC);
+
+    if (bridge_port_id == SAI_NULL_OBJECT_ID && bv_id == SAI_NULL_OBJECT_ID)
+    {
+        for (auto itr = m_entries.begin(); itr != m_entries.end();)
+        {
+            auto curr = itr++;
+            if (curr->second.type.find(fdb_type) != std::string::npos)
```

Comment:
I think this check is redundant. afaik, the fdb_type is always populated.

Reply:
These entries might be static as well, so i think it's required to check the fdb_type before clearing them

Reply:
so you are not checking for static vs dynamic here. my point is, one or other is always true and never be npos, right?

Reply:
May be i'm missing something. what do we miss with the following?

```
            if (curr->second.type.find(fdb_type) != std::string::npos)
            {
                if (curr->first.mac == mac || mac == flush_mac)
                {
                    clearFdbEntry(curr->first.mac, curr->first.bv_id, fdb_type, curr->first.port_name);
                }
            }
```

to 
```

                if (curr->first.mac == mac || mac == flush_mac)
                {
                    clearFdbEntry(curr->first.mac, curr->first.bv_id, fdb_type, curr->first.port_name);
                }
```

Reply:
We're basically checking if the entry type stored in m_entries is equivalent to what was received by flush. 

i.e. only proceed forward in these (static == static || dynamic == dynamic || dynamic == dynamic_local) scenarios

Reply:
Got it. In the flush, can user specify fdb_type? If not, is it correct that flush notification would be for all types?

Reply:
[Link](https://github.com/opencomputeproject/SAI/blob/master/inc/saifdb.h#L222) User can specify the flush type. Looks like there can also be a SAI_FDB_FLUSH_ENTRY_TYPE_ALL along with SAI_FDB_FLUSH_ENTRY_TYPE_STATIC/SAI_FDB_FLUSH_ENTRY_TYPE_DYNAMIC

Reply:
I'll update to handle the SAI_FDB_FLUSH_ENTRY_TYPE_ALL case as well.

Reply:
Sry, the flush notif will have either STATIC or DYNAMIC, so "all" case need not be handled.

Reply:
EDIT: Updated, remove reading the SAI_FDB_ENTRY_ATTR_TYPE attribute and cleared only non-static entries